### PR TITLE
geomodel: Fix dependencies

### PR DESCRIPTION
**Pull request by paulgessinger**
This PRs makes the following changes

- Change existing dependency on `hdf5` to `hdf5+cxx`
- Add dependency on `hdf5` when building `+fullsimlight`
- GeoModel can build with Qt6 **if it ships with the Qt5 compatibility layer**. The Qt6 spec (`qt-base`) does not have this ability, and there is a Qt5 spec (`qt`), so I'm switching this spec over to just use Qt5 for now. `qt` doesn't have the `+widgets` variant, but the build seems to build without.
- GeoModel visualization requires `coin3d` to build

### CODE REVIEW

**Review by stephenswat — APPROVED** [review]
LGTM.

**Comment by wdconinc** [comment]
> if it ships with the Qt5 compatibility layer.

Do you mean `qt-5compat`?

**Comment by paulgessinger** [comment]
@wdconinc I guess (not experienced with Qt). But even with that included in the build, the `find_package` call looking for `Qt6Core5Compat` does not find anything. It seems to resolve to the correct `Qt6Config.cmake` file generated by the `qt-base` build.

I guess I don't understand how `qt-5compat` works / should work. I also can't say I understand the qt package structure with `qt` (<=5) and `qt-base` (>=6) + `qt-5compat`.

I'm also finding (don't know why I didn't see this earlier) that GeoModel also wants "SoQt" through Coin3D, which I couldn't find any reference to in the respective packages.

As I'm not extremely interested in adding a package for this, or adding it to Coin3D, I'm considering just removing the `visualization` variant from this package, since it doesn't build anyway.

**Comment by wdconinc** [comment]
> I also can't say I understand the qt package structure with `qt` (<=5) and `qt-base` (>=6) + `qt-5compat`.

With qt<=5 we had everything in one large package, which was leading to maintainability issues (the package.py is enormous). With qt>=6 we split it into different packages for different components, similar to how qt structures it in their repositories.

Leaving out visualization is of course fine, or even limiting it to qt5.

**Comment by paulgessinger** [comment]
@wdconinc Ok I see. The problem however seems to be that GeoModel does (as ostensibly documented by Qt)

```cmake
find_package(Qt6 REQUIRED COMPONENTS Core5Compat)
```

which fails with the separate `qt-5compat` package. Is that supposed to work?

**Comment by paulgessinger** [comment]
I opened https://github.com/spack/spack/pull/47443 to add soqt now.

**Comment by wdconinc** [comment]
This will need a rebase. I can't convince it to use a Qt6 installation (or the compatibility layer isn't detected).

**Comment by paulgessinger** [comment]
@wdconinc Rebased now.

I believe the issue is that GeoModel tries to find Qt6 using

```cmake
find_package( Qt${QT_VERSION} COMPONENTS Core Gui Widgets Sql PrintSupport OpenGL Core5Compat  REQUIRED )
```

where `Core5Compat` is a component in the `find_package` call. With the `qt-5compat` spack package, what is the correct way to locate the compatibility component using CMake? Is it a separate `find_package` call?

**Comment by wdconinc** [comment]
> With the qt-5compat spack package, what is the correct way to locate the compatibility component using CMake? Is it a separate find_package call?

It should be sufficient to just depend on `qt-5compat` but I don't think there is any working example that does that already. I thought it worked in #45300 but I have to verify that again when I have a chance to get back to that PR.

**Comment by paulgessinger** [comment]
@wdconinc So should we go ahead with this PR?

**Comment by wdconinc** [comment]
@paulgessinger If this currently works with `qt` (not `qt-5compat`), then I think it's fine to go ahead with this. Please confirm and I'll merge.

I found some time to tinker with `qt-tools` and some other dependencies of the `paraview ^qt-base` stack, so I could imagine this is not too far away from working. But I don't want to hold up the other aspects.

**Review by wdconinc — APPROVED** [review]

**Comment by wdconinc** [comment]
Nevermind, I see it is requiring `qt` now, so I'll go ahead and merge right away.

**Pull request merged**
